samples/spring-transfer-workflow/src/main/java/com/example/transfer/TransferWorkflowEntity.java

```diff
@@ -0,0 +1,181 @@
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+// tag::class[]
+@EntityType("transfer") // <2>
+@EntityKey("transferId") // <3>
+@RequestMapping("/transfer/{transferId}") // <4>
+public class TransferWorkflowEntity extends WorkflowEntity<TransferState> { // <1>
+  // end::class[]
+
+  // tag::start[]
+  public record Withdraw(String from, int amount) { // <4>
+  }
+
+  // end::start[]
+
+  // tag::definition[]
+  public record Deposit(String to, int amount) { // <4>
+  }
+
+  // end::definition[]
+
+
+  private static final Logger logger = LoggerFactory.getLogger(TransferWorkflowEntity.class);
+
+  private final String fraudDetectionStepName = "fraud-detection";
+
+  // tag::definition[]
+  private final String withdrawStepName = "withdraw"; // <1>
+  private final String depositStepName = "deposit"; // <1>
+
+  // end::definition[]
+
+  final private KalixClient kalixClient;
+
+  public TransferWorkflowEntity(KalixClient kalixClient) {
+    this.kalixClient = kalixClient;
+  }
+
+  // tag::definition[]
+  @Override
+  public Workflow<TransferState> definition() {
+    Step withdraw =
+        step(withdrawStepName) // <1>
+            .call((Withdraw cmd) -> kalixClient.patch("/wallet/" + cmd.from() + "/withdraw/" + cmd.amount(), String.class)) // <2>
+            .andThen(this::moveToDeposit);
+
+    Step deposit =
+        step(depositStepName) // <1>
+            .call((Deposit cmd) -> kalixClient.patch("/wallet/" + cmd.to() + "/deposit/" + cmd.amount(), String.class)) // <5>
+            .andThen(this::finishWithSuccess);
+
+    return workflow() // <7>
+        .addStep(withdraw)
+        .addStep(deposit);
+  }
+
+  private Effect.TransitionalEffect<Void> moveToDeposit(String response) {
+    TransferState transferState = currentState();
+    TransferState successfulWithdrawal = transferState.asSuccessfulWithdrawal();
+    Deposit depositInput = new Deposit(transferState.transfer().to(), transferState.transfer().amount()); // <4>
+    return effects()
+        .updateState(successfulWithdrawal)
+        .transition(depositInput, depositStepName); // <3>
+  }
+
+  private Effect.TransitionalEffect<Void> finishWithSuccess(String response) {
+    TransferState completed = currentState().asCompleted();
+    return effects()
+        .updateState(completed)
+        .end(); // <6>
+  }
+  // end::definition[]
+
+  // tag::start[]
+  @PutMapping
+  public Effect<Message> startTransfer(@RequestBody Transfer transfer) {
+    if (transfer.amount() <= 0) {
+      return effects().error("transfer amount should be greater than zero"); // <1>
+    } else if (currentState() == null) {
+
+      TransferState initialState = new TransferState(transfer); // <2>
+
+      Withdraw withdraw = new Withdraw(transfer.from(), transfer.amount());
+
+      return effects()
+          .updateState(initialState) // <3>
+          .transition(withdraw, withdrawStepName) // <4>
+          .thenReply(new Message("transfer started")); // <5>
```

Comment:
What about this?

```java
return effects()
    .updateState(new TransferState(transfer)) 
    .transition(new Withdraw(transfer.from(), transfer.amount()), "withdraw") 
    .thenReply(new Message("transfer started")); 
```

For documentation, is nice if we have it more compact. 

Also nice to have it more self-contained. So, "withdraw" instead of `withdrawStepName`. The rational here is that, if we use a variable, we need to explain it. We need to first show that we defined it and then use it. 

It's certainly a best-practice, but we can leave this to the user since it's a general best-practice. IMO, we should only suggest best-practice in domains that may not be familiar for the user. For general stuff, better if let the users use their own judgement. It saves of time and make it easier to document.